Automattic/vip-go-mu-plugins
# Updates the way vip-search versioning information is stored

## Description
* Stores global indexables to site_option (one per network)
* Stores local indexables to option (one per site)
* Note: for single-site instalation site_option falls back to option.



## Changelog Description

### Updates the way vip-search versioning information is stored

The versioning information for vip-search index needed to be updated to address global indexables (e.g. User), and different versioning across sites in a multisite environment.

This change stores global indexable separately - one per network. While making sure non-global versions are stored one per site.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

* Activate user indexables `lando wp elasticpress activate-feature users`
* Add version `lando wp vip-search index-versions add user`
* Check the GLOBAL option: `lando wp eval "var_dump(get_site_option('vip_search_global_index_versions'));"`
* Should have users versions only
* Check the LOCAL option: `lando wp eval "var_dump(get_option('vip_search_index_versions'));"`
* Should have posts versions only
* Check that retrieving version still works as well e.g. `lando wp vip-search index-versions list post`
